plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.metastore.HiveMetastore;
+import io.trino.plugin.hive.containers.Hive4MinioDataLake;
+import io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore;
+import io.trino.plugin.hive.s3.S3HiveQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.containers.Minio;
+import io.trino.testing.minio.MinioClient;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import static io.trino.plugin.hive.TestingThriftHiveMetastoreBuilder.testingThriftHiveMetastoreBuilder;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static org.junit.jupiter.api.Assumptions.abort;
+
+@Execution(ExecutionMode.SAME_THREAD) // Custom hive4 image doesn't support running concurrent queries
+class TestHive4OnDataLake
+        extends BaseTestHiveOnDataLake
+{
+    private String bucketName;
+    private Hive4MinioDataLake hiveMinioDataLake;
+    private HiveMetastore metastoreClient;
+
+    @Override
```

Comment:
move common code from createQueryRunner() as well to BaseTestHiveOnDataLake

Reply:
done